### ISSUE TITLE
[swift][swift3] Potential naming conflict with ErrorResponse

### ISSUE THREAD

**Issue by philippelatulippe** [swagger-api/swagger-codegen#4208]
##### Description

The PR #3624 introduced an enum called "ErrorResponse" to support returning HTTP error information. This is a great feature. But "ErrorResponse" is a fairly common definition in swagger specs, as you can see by searching github for yaml files containing it. There's even some snippets posted on this issue tracker that have an definition called "ErrorResponse". Code generated with the swift or swift3 generator won't compile if the spec contains a definition called "ErrorResponse", as the generated class and the enum will conflict.

The enum is in this file: [modules/swagger-codegen/src/main/resources/swift3/Models.mustache](https://github.com/swagger-api/swagger-codegen/blob/master/modules/swagger-codegen/src/main/resources/swift3/Models.mustache)

It would be great if *ErrorResponse* could be renamed to something more verbose, or somehow namespaced.

##### Swagger-codegen version

master

##### Swagger declaration file

```
swagger: '2.0'
info:
  version: "0.0.0"
  title: Mini
paths:
  /persons:
    get:
      responses:
        400:
          description: Error
          schema:
            $ref: "#/definitions/ErrorResponse"
definitions:
  ErrorResponse:
    type: object
    properties:
      errorCode:
        type: integer
```

##### Command line used for generation

    swagger-codegen-dev generate -l swift3 -i swagger.yaml -o ~/Code/MiniClient

##### Steps to reproduce

1. Use the yaml file above.
2. Run the generation command above.
3. Add the generated code to an xcode project.
4. Build

You will get the error
> Invalid redeclaration of 'ErrorResponse'

**Comment by wing328** [comment]
@philippelatulippe we will escape `ErrorResponse` by appending `Model` at the beginning, i,e. `ModelErrorResponse`. Other language generator has applied similar solution.

**Comment by wing328** [comment]
Closed via #4221. Please pull the latest master to give it a try.
